pkg/core/xds/merge.go

```diff
@@ -42,18 +44,123 @@ func MergeConfs(confs []interface{}) (interface{}, error) {
 		return nil, err
 	}
 
+	valueResult := reflect.ValueOf(result)
 	// clear appendable slices, so we won't duplicate values of the last conf
-	clearAppendSlices(reflect.ValueOf(result))
+	clearAppendSlices(valueResult)
 	for _, conf := range confs {
 		// call .Elem() to unwrap interface{}
-		appendSlices(reflect.ValueOf(result), reflect.ValueOf(&conf).Elem())
+		appendSlices(valueResult, reflect.ValueOf(&conf).Elem())
 	}
 
-	v := reflect.ValueOf(result).Elem().Interface()
+	if err := handleMergeByKeyFields(valueResult); err != nil {
+		return nil, err
+	}
+
+	v := valueResult.Elem().Interface()
 
 	return v, nil
 }
 
+type acc struct {
+	Key      interface{}
+	Defaults []interface{}
+}
+
+const (
+	defaultFieldName = "Default"
+	xdsMergeTag      = "xdsMerge"
```

Comment:
I wouldn't bring `xds` name here, it doesn't really relate to xDS

Reply:
Yeah just picked it cause the package. I'll call it `policyMerge`